### ISSUE TITLE
Some keyboard shortcuts not working

### ISSUE THREAD

**Issue by gchriz** [WikidPad/WikidPad#9]
A few keyboard shortcuts are not working as expected / at all, depending on the operating system and wxPython version.

I created a table with all standard WikidPad shortcuts and my test results on wiki page 
[KeyboardShortcuts](https://github.com/WikidPad/WikidPad/wiki/KeyboardShortcuts).

I'll try to find out more details soon.

It would be helpful if other users could test as well and report their results.

**Comment by butscher** [comment]
Regarding the issue on Windows (which I can't reproduce) this is probably either some tool program running in the background which reacts on the key strokes (if it behaves nicely it shows at least a tray icon) or the keyboard shortcut is defined as a property of the program entry in the start menu. 

In the latter case find the program in start menu, right click, choose "Properties". There is an entry where a keyboard shortcut can be set to start the program.

**Comment by gchriz** [comment]
You are correct. I found the reasons on the Windows testing systems and fixed the table accordingly.
On Windows all shortcuts are working now.

**Comment by gchriz** [comment]
On newer Linux (wxPython 3.x) two problems remain:

1.) Ctrl-Alt-LETTER combinations fail, probably due to wxPython and/or GTK issues

2.) Ctrl-Up and Ctrl-Down are only working as expected (View parents/children) if the focus is **not** in the editor! If the focus is in the editor they perform a Scroll-Up/Scroll down

I'll do more tests with these.

**Comment by xkjq** [comment]
I think I fixed this a while ago (its to do with different versions of wxpython using different accelerator combinations if I remember correctly) back in my old repo (https://bitbucket.org/xkjq/wikidpad_svn). 

Problem is that repo got into a complete mess, I've got some spare time over the next few days so I'm going to look at migrating the (useful) changes back to a fork of this one.

**Comment by butscher** [comment]
> 2.) Ctrl-Up and Ctrl-Down are only working as expected (View parents/children) if the focus is not in the editor (e.g. in the search field instead)! If the focus is in the editor they perform a Scroll-Up/Scroll down

It seems that the order of key shortcut processing changed with newer wxPython versions. As a workaround you can modify the code. At the end of file "lib/EnhancedScintillaControl.py" there is a large list named "_DEFAULT_STC_KEYS". All these keys are defined to be processed by the Scintilla editor. You can delete or comment out the lines

`(wx.stc.STC_KEY_DOWN, wx.stc.STC_SCMOD_CTRL, wx.stc.STC_CMD_LINESCROLLDOWN),`

and

`(wx.stc.STC_KEY_UP, wx.stc.STC_SCMOD_CTRL, wx.stc.STC_CMD_LINESCROLLDOWN),`

I haven't tested it yet but it should allow the shortcuts to be processed correctly.

**Comment by gchriz** [comment]
On 17.03.2016 at 02:02 Michael Butscher wrote:

>> 2.) Ctrl-Up and Ctrl-Down not working as expected
> 
> You can delete or comment out the lines
>   (wx.stc.STC_KEY_DOWN, wx.stc.STC_SCMOD_CTRL, ...
> and
>   (wx.stc.STC_KEY_UP, wx.stc.STC_SCMOD_CTRL, ...


Unfortunately that doesn't change anything.
